x/claims/genesis.go

```diff
@@ -38,16 +39,29 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, data types.GenesisState) {
 			InitialClaimableAmount: claimsRecord.InitialClaimableAmount,
 			ActionsCompleted:       claimsRecord.ActionsCompleted,
 		}
-		k.SetClaimsRecord(ctx, addr, cr)
 
-		sumClaims = sumClaims.Add(cr.InitialClaimableAmount)
+		if len(cr.ActionsCompleted) != len(types.Action_name)-1 {
+			panic(fmt.Errorf("invalid actions completed length for address %s", claimsRecord.Address))
+		}
+
+		initialClaimablePerAction := claimsRecord.InitialClaimableAmount.Quo(numActions)
+
+		for _, actionCompleted := range cr.ActionsCompleted {
+			if !actionCompleted {
+				// NOTE: only add the initial claimable amount per action for the ones that haven't been claimed
+				sumUnclaimed = sumUnclaimed.Add(initialClaimablePerAction)
```

Comment:
You have to use:
https://github.com/tharsis/evmos/blob/f5dfed87da17ee61c1338f8d70b27f26f3d1b17a/x/claims/keeper/claim.go#L11

If you don't, genesis export will fail on decay of the claims balances. It was in the previous commit for a reason.

Reply:
Agreed... if we can incorporate this now.. it will prevent us from having to remember that detail should we ever need to export state again in the next few months

Reply:
@fedekunze ^^ I'll make an updated PR to this effect